rudder-server-root/SOURCES/Makefile

```diff
@@ -16,18 +16,86 @@
 #
 #####################################################################################
 
-.DEFAULT_GOAL := localdepends
+.DEFAULT_GOAL := build
 
 RUDDER_VERSION_TO_PACKAGE = <put Rudder version or version-snapshot here>
 
+# Autodetect presence of gnu tools
+TAR := $(shell type gtar >/dev/null 2>&1 && echo gtar || echo tar)
+PATCH := $(shell type gpatch >/dev/null 2>&1 && echo gpatch || echo patch)
+SED := $(shell type gsed >/dev/null 2>&1 && echo gsed || echo sed)
+INSTALL := $(shell type ginstall >/dev/null 2>&1 && echo ginstall || echo install)
+CP_A := cp -rp
+FIND := find
+
+# default: use https for urls
+ifeq (false,$(USE_HTTPS))
+HTTPS:=http
+else
+HTTPS:=https
+endif
+
+# GENERIC_GET: do not remove this line
+# This is a generic get code that is common to many of our Makefiles.
+# The tag is here to find them.
+# If you change this code, change all the places where you can find this tag too.
+# It is duplicated because it is the code that is used to manage dependency
+# to other repositories, hence allowing deduplication of other code.
+# Autodetect wget, curl or fetch usage and proxy configuration
+# Usage: $(GET) <destination_file> <url>
+PROXY_ENV = $(if $(PROXY), http_proxy=$(PROXY) ftp_proxy=$(PROXY))
+# No accept encoding to prevent the webserver form altering the distributed file
+WGET = wget -q --header="accept-encoding:" -O
+CURL = curl -s -L -o
+FETCH = fetch -q -o
+ifneq (,$(wildcard /usr/bin/curl))
+_GET = $(PROXY_ENV) $(CURL)
+GET_OPT = --curl
+CPAN_OPT = --curl --no-wget
+else
+ifneq (,$(wildcard /usr/bin/fetch))
+_GET = $(PROXY_ENV) $(FETCH)
+else
+_GET = $(PROXY_ENV) $(WGET)
+GET_OPT = --wget
+CPAN_OPT = --wget
+endif
+endif
+# We are using sha1 as sha256 is not supported on RHEL3
+# Not using the sha1sum command as it is not available on AIX
+# Pass an empty string to skip hash check
+GET=get() { $(_GET) "$$1.part" "$$2" && { openssl dgst -sha1 "$$1.part" | grep -q "$$3" || { echo "Wrong checksum, aborting"; exit 1; }; } && mv "$$1.part" "$$1"; }; get
+
+# Defauth PATHs
+RUDDER_DIR = /opt/rudder
+DESTDIR = $(CURDIR)/target
+INSTALL_DIR = $(DESTDIR)$(RUDDER_DIR)
+
 localdepends: ./rudder-sources
 
-./rudder-sources.tar.bz2:
-	$(WGET) -O rudder-sources.tar.bz2 http://www.rudder-project.org/archives/rudder-sources-${RUDDER_VERSION_TO_PACKAGE}.tar.bz2
+rudder-sources.tar.bz2:
+        $(GET) rudder-sources.tar.bz2 $(HTTPS)://www.rudder-project.org/archives/rudder-sources-$(RUDDER_VERSION_TO_PACKAGE).tar.bz2 ""
+
+rudder-sources: rudder-sources.tar.bz2
+        bunzip2 < rudder-sources.tar.bz2 | $(TAR) xf -
+        mv rudder-sources-$(RUDDER_VERSION_TO_PACKAGE)*/ rudder-sources/ # the '*' is needed for nightly release
+        touch rudder-sources # to make it younger than the bzip source file
+
+build:
+	true
+
+install:
+	# Init files
```

Comment:
Since this is a double negation, put a comment to say what it means for the default value

Reply:
could be better, like "default is false -> use init"